jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/App.java

```diff
@@ -117,9 +132,16 @@ public String getContextPath()
         return _context == null ? null : _context.getContextPath();
     }
 
-    public String getEnvironment()
+    /**
+     * Get the environment name.
+     * If the property "environment" exists, then that is returned as the environment, otherwise
+     * the {@link DeploymentManager#getDefaultEnvironmentName()} is returned.
+     * @return The {@link org.eclipse.jetty.util.component.Environment} name for the application.
+     */
+    public String getEnvironmentName()
     {
-        return _environment;
+        String name = getProperties().get("environment");
```

Comment:
Is this something that should be known in the constructor and immutable thereafter?

Reply:
properties can be mutated after construction